src/bcs/boundary_types.jl

```diff
@@ -104,16 +104,15 @@ function generate_boundary_matching_rules(s, orders)
     upperboundary(x) = last(s.axies[x])
 
     # Rules to match boundary conditions on the lower boundaries
-    lower = reduce(vcat, [_boundary_rules(s, orders, x, lowerboundary(x)) for x in s.x̄])
+    lower = Dict([operation(u) => Dict([x => _boundary_rules(s, orders, u, x, lowerboundary(x)) for x in params(u, s)]) for u in s.ū])
```

Comment:
Why dict? Do we not use the ordering?

Reply:
This is needed to seperate between rules for each variable and boundary, to correctly match periodic boundaries. I could do it with a smarter indexing function though.

I moved over to a lot of dicts to enable more flexible depvars with different argument signatures, again I could do this with a smart indexing function but I thought this was a good first pass.

Reply:
okay that's totally fine.